gensim/corpora/textcorpus.py

```diff
@@ -98,28 +98,54 @@ def get_texts(self):
                 else:
                     yield utils.tokenize(line, lowercase=True)
 
-    def sample_texts(self, n):
+    def sample_texts(self, n, seed=None, length=None):
         """
-        Yield n random texts from the corpus without replacement.
+        Yields n random documents from the corpus without replacement.
 
-        Given the the number of remaingin elements in stream is remaining and we need
-        to choose n elements, the probability for current element to be chosen is n/remaining.
+        Given the number of remaining documents in corpus, we need to choose n elements.
+        The probability for current element to be chosen is n/remaining.
         If we choose it, we just decreese the n and move to the next element.
+        Computing corpus length may be a costly operation so you can use optional paramter
+        length instead.
+
+        Args:
+            n (int): number of documents we want to sample.
+            seed (int|None): if specified, use it as a seed for local random generator.
+            length (int|None): if specified, use it as guess of corpus length.
+
+        Yeilds:
+            list[str]: document represented as list of tokens. See get_texts method.
+
+        Raises:
+            ValueError: then n is invalid or length was set incorrectly.
```

Comment:
"then" => "when"?

Also, what does "incorrectly" mean? Above we say `length` is "a guess", so how can it be incorrect?